src/App.tsx

```diff
@@ -7,33 +9,91 @@ function getRandomName(): string {
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type Props = {};
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+type State = {
+  time: string;
+  oldName: string | {},
+  newName: string | {},
+  hasClock: boolean,
+};
+
+export class App extends React.Component<Props, State> {
+  state = {
+    time: '',
+    oldName: 'Clock-0',
+    newName: 'Clock-0',
+    hasClock: true,
+  };
+
+  timerID = 0;
+
+  componentDidMount() {
+    window.setInterval(() => {
+      this.setState({ time: new Date().toLocaleTimeString() });
+    }, 1000);
+
+    this.timerID = window.setInterval(() => {
+      this.setState(prevState => (
+        { oldName: prevState.newName }
+      ));
+
+      this.setState({ newName: getRandomName() });
+    }, 3300);
+
+    document.addEventListener('mousedown', this.clockShow);
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentDidUpdate(prevProps: {}, prevState: {}) {
+    if (this.state.hasClock === true
+      && this.state.oldName !== prevState
+      && this.state.newName !== prevState
+      && this.state.oldName !== prevProps
+      && this.state.newName !== prevProps) {
+      console.info(`Renamed from ${this.state.oldName} to ${this.state.newName}`);
+    }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    if (this.state.hasClock === true
+      && this.state.time !== prevState
+      && this.state.time !== prevProps) {
```

Comment:
you don't need to check `prevProps` here
```suggestion
    if (this.state.hasClock === true && this.state.time !== prevState.time) {
```